### PR TITLE
adds the start of the api functionality with the endpoint GET /api/ca…

### PR DESCRIPTION
**Pull request by Hm255**
GET categories

### CODE REVIEW

File: db/connection.js [hunk 1]
```diff
@@ -8,5 +8,8 @@ require('dotenv').config({
 if (!process.env.PGDATABASE) {
   throw new Error('PGDATABASE not set');
 }
+if(process.env.PGDATABASE){
```
Comment (two-triangles) [inline db/connection.js]: We don't need to check the PGDATABASE variable immediately after checking whether it doesn't exist. We can check that it doesn't exist and throw an error accordingly. 

If it does exist, then we would never enter that conditional logic block and we would carry on to export our pool of connections to the database

File: app.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const express = require("express");
+const { getCategories } = require("./controller/controller");
+const app = express();
+app.use(express.json());
+
+app.get("/api/categories", getCategories);
+
+app.use((req, res, next) => {
+  console.log(req.body);
```
Comment (two-triangles) [inline app.js]: What is the purpose of this middleware function?

File: controller/controller.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const {fetchCategories} = require('../model/model')
+//handling sql queries and directing them to an output in the controller
+
+
+
+exports.getCategories = (req, res) => {      //takes category data defined in the model and sends it through
+  console.log('found the controller')
```
Comment (two-triangles) [inline controller/controller.js]: once finished debugging, remember to remove console.logs

**Comment by two-triangles** [comment]
I have left some comments, please amend and drop me a message once you have done so and I will have another look over before merging

File: db/connection.js (hunk 1 above)
Comment (Hm255) [reply db/connection.js]: originally I had muddled up my program a lot so I wasn't sure which parts were entirely functional or not, the visual cue helped me. I have added console logs for other parts as well and I will remove them

File: controller/controller.js (hunk 3 above)
Comment (Hm255) [reply controller/controller.js]: Question, am I alright in adding comments to guide me about MVC ordering inside the model/controller?

Comment (two-triangles) [reply controller/controller.js]: Yes this is fine, but once you have that task completed, they need to be deleted.

File: db/connection.js (hunk 1 above)
Comment (two-triangles) [reply db/connection.js]: okay, well we can delete that second if statement as we will have already checked for the variable existence in the previous conditional statement

File: db/connection.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-const { Pool } = require('pg');
-const ENV = process.env.NODE_ENV || 'development';
+const { Pool } = require('pg'); 
+const ENV = process.env.NODE_ENV || 'test';
```
Comment (MRombi) [inline db/connection.js:2]: This is the correct one: const { Pool } = require('pg');
const ENV = process.env.NODE_ENV || 'development';

Change 'test' with 'development' please 🙂

File: model/model.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+const db = require("../db/connection");
+
+
+exports.fetchCategories = () => {
+    return db.query(`SELECT * FROM categories`).then(({ rows }) => {
+        console.log(rows)
```
Comment (MRombi) [inline model/model.js:6]: Remove this console.log please 🙂

**Review by MRombi — COMMENTED** [review]
Well done on this task, there are a few minor adjustments to be made but after that you are good to merge 😄

**Pull request closed without merging**